### ISSUE TITLE
label-content-name-mismatch wrongly includes invisible text

### ISSUE THREAD

**Issue by jribbens** [dequelabs/axe-core#4678]
### Product

axe-core

### Product Version

4.10.2

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation
Given the following HTML code, axe _should not_ report an label-content-name-mismatch violation:
```html
<a aria-label="deque are great" href="https://www.deque.com/">
  deque <span class="visually-hidden">labs</span> are great
</a>
```
Given the following HTML code, axe _should_ report an label-content-name-mismatch violation:
```html
<a href="https://www.deque.com/">
  deque <span class="visually-hidden">labs</span> are great
</a>
```

#### Actual
It emits a diagnostic on the code which is correct, and does not emit one on the code which is incorrect.

#### How to Reproduce
See above. Or also: https://unequivocal.eu/label-content-name-mismatch.html

#### Additional context
The rule is supposed to be comparing the accessible name with the _visible_ text of the item, but it seems to be comparing all of the text. Also it only checks elements with an `aria-label` or `aria-labelledby`, which is incorrect as the rule can fail even without an ARIA label (i.e. the second example above). See https://www.w3.org/WAI/WCAG21/Techniques/failures/F96 for example.

**Comment by dylanb** [comment]
@jribbens what is the CSS for the `visually-hidden` class in both cases?

**Comment by WilcoFiers** [comment]
Thank you for reporting this @jribbens. This is an interesting case. I certainly agree the first examples shouldn't fail. I'm not entirely sure we can reliably fail the second scenario. We'll have to think bout that when we're looking at how to solve for this.

**Comment by WilcoFiers** [comment]
For later reference. I've done some testing of this today, and at least MacOS's Voice Control seems to require announcing the full accessible name. So hidden texts are kind of a problem.

**Comment by chutchins25** [comment]
The false positive here — the rule including the visually-hidden text — was fixed in #5096, which switched the check to compare against `visibleVirtual` (visible-on-screen text) instead of `subtreeText`. Verified on axe 4.12.1: the first example (`aria-label="deque are great"` with a `.visually-hidden` "labs") now passes. #5291 adds the regression tests that #5096 didn't include, so this can close.

The second point — that the rule only applies to elements with `aria-label`/`aria-labelledby`, so name-from-content elements (the second example) aren't evaluated at all — is the same "axe under-fails relative to ACT" gap tracked in #5203 (consistency with ACT rule 2ee8b8). I've flagged the name-from-content case there so it's handled alongside the rest of the ACT alignment.

**Comment by straker** [comment]
The first false positive has been fixed in https://github.com/dequelabs/axe-core/pull/5096. The second false negative is being tracked separately in https://github.com/dequelabs/axe-core/issues/5339. Going to close as completed.
